frontend/components/ProfessorGradingModal.tsx

```
@@ -0,0 +1,692 @@
+                            </span>
+                            {!isUser && !isSystem && msg.persona_name && (
+                              <Badge variant="secondary" className="text-xs bg-white/90 backdrop-blur-sm text-gray-800 border border-gray-300/50 shadow-sm font-medium">
+                                {msg.persona_role || getPersonaRole(msg.persona_name) || msg.persona_name}
+                              </Badge>
+                            )}
+                          </div>
+                          <div className="text-sm whitespace-pre-wrap leading-relaxed">
+                            {msg.content.split('\n').map((line, index) => {
+                              const boldFormatted = line.replace(/\*\*(.*?)\*\*/g, '<strong class="font-semibold">$1</strong>')
+                              return (
+                                <div key={index} dangerouslySetInnerHTML={{ __html: boldFormatted }} />
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)